From 164758793a6e27826543b6382ac66817c991ddc4 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Thu, 25 Jan 2024 17:24:39 +0100
Subject: [PATCH 472/841] refactor(st-pmic): remove useless structure fields

Cleanup useless structure fields in order to
reduce code size.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 9302babde433198dc69725848f36f2be163ec072 refactor(st-pmic): remove useless structure fields]
Change-Id: If5ccfda63d27dbdabf929c2fa7542ba8663070ef
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/354521
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/st/pmic/stpmic2.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/drivers/st/pmic/stpmic2.c b/drivers/st/pmic/stpmic2.c
index be9e86420..075f62dcf 100644
--- a/drivers/st/pmic/stpmic2.c
+++ b/drivers/st/pmic/stpmic2.c
@@ -28,13 +28,10 @@ struct regul_struct {
 	uint8_t volt_shift;
 	uint8_t en_cr;
 	uint8_t alt_en_cr;
-	uint8_t alt_volt_cr;
-	uint8_t pwrctrl_cr;
 	uint8_t msrt_reg;
 	uint8_t msrt_mask;
 	uint8_t pd_reg;
 	uint8_t pd_val;
-	uint8_t pd_mask;
 	uint8_t ocp_reg;
 	uint8_t ocp_mask;
 };
@@ -98,13 +95,10 @@ static const uint16_t refddr_volt_table[] = {
 	.en_cr			= ID ## _MAIN_CR2, \
 	.volt_cr		= ID ## _MAIN_CR1, \
 	.alt_en_cr		= ID ## _ALT_CR2, \
-	.alt_volt_cr		= ID ## _ALT_CR1, \
-	.pwrctrl_cr		= ID ## _PWRCTRL_CR, \
 	.msrt_reg		= BUCKS_MRST_CR, \
 	.msrt_mask		= ID ## _MRST, \
 	.pd_reg			= pd, \
 	.pd_val			= ID ## _PD_FAST, \
-	.pd_mask		= ID ## _PD_MASK, \
 	.ocp_reg		= FS_OCP_CR1, \
 	.ocp_mask		= FS_OCP_ ## ID, \
 }
@@ -117,13 +111,10 @@ static const uint16_t refddr_volt_table[] = {
 	.en_cr			= ID ## _MAIN_CR, \
 	.volt_cr		= ID ## _MAIN_CR, \
 	.alt_en_cr		= ID ## _ALT_CR, \
-	.alt_volt_cr		= ID ## _ALT_CR, \
-	.pwrctrl_cr		= ID ## _PWRCTRL_CR, \
 	.msrt_reg		= LDOS_MRST_CR, \
 	.msrt_mask		= ID ## _MRST, \
 	.pd_reg			= LDOS_PD_CR1, \
 	.pd_val			= ID ## _PD, \
-	.pd_mask		= ID ## _PD, \
 	.ocp_reg		= FS_OCP_CR2, \
 	.ocp_mask		= FS_OCP_ ## ID, \
 }
@@ -135,13 +126,10 @@ static const uint16_t refddr_volt_table[] = {
 	.en_cr			= ID ## _MAIN_CR, \
 	.volt_cr		= ID ## _MAIN_CR, \
 	.alt_en_cr		= ID ## _ALT_CR, \
-	.alt_volt_cr		= ID ## _ALT_CR, \
-	.pwrctrl_cr		= ID ## _PWRCTRL_CR, \
 	.msrt_reg		= BUCKS_MRST_CR, \
 	.msrt_mask		= ID ## _MRST, \
 	.pd_reg			= LDOS_PD_CR2, \
 	.pd_val			= ID ## _PD, \
-	.pd_mask		= ID ## _PD, \
 	.ocp_reg		= FS_OCP_CR1, \
 	.ocp_mask		= FS_OCP_ ## ID, \
 }
-- 
2.47.2

